src/_pytest/assertion/rewrite.py

```diff
@@ -166,6 +170,38 @@ def find_module(self, name, path=None):
         self.modules[name] = co, pyc
         return self
 
+    def _early_rewrite_bailout(self, name, state):
+        """
+        This is a fast way to get out of rewriting modules. Profiling has
+        shown that the call to imp.find_module (inside of the find_module
+        from this class) is a major slowdown, so, this method tries to
+        filter what we're sure won't be rewritten before getting to it.
+        """
+        if self.session is not None:
+            # Make something as c:/projects/my_project/path.py ->
+            #     ['c:', 'projects', 'my_project', 'path.py']
+            parts = str(self.session._initialpaths).split(os.path.sep)
```

Comment:
Isn't `session._initialpaths` a list? If so this should be something like:

```python
for path in self.session._initialpaths:
    name = os.path.splitext(os.path.basename(str(path)))[0]
    self._basenames_to_check_rewrite.add(name)
```

Also this is being done every time `_early_rewrite_bailout` is being called, it should probably be done in `set_session` instead.

Reply:
You're right, `session._initialpaths` is a container (actually, a `set`). Will fix it (tests failed because of it). 

I'm doing it here because it's only filled during collect (thus not available in `set_session`) -- I can add a flag so that it's only done once though (it should be immutable for that to work, so, I'll change it to be a frozenset so that we can rely on that).